dartagnan/src/main/java/com/dat3m/dartagnan/verification/solving/AssumeSolver.java

```diff
@@ -33,46 +30,49 @@ private AssumeSolver(SolverContext c, ProverEnvironment p, VerificationTask t) {
         task = t;
     }
 
-    public static Result run(SolverContext ctx, ProverEnvironment prover, VerificationTask task)
+    public static AssumeSolver of(SolverContext ctx, ProverEnvironment prover, VerificationTask task)
             throws InterruptedException, SolverException, InvalidConfigurationException {
-        return new AssumeSolver(ctx, prover, task).run();
+        AssumeSolver s = new AssumeSolver(ctx, prover, task);
+        s.run();
+        return s;
```

Comment:
Method should get renamed, in particular since the model checker not only gets created but it also runs!